### PR TITLE
[FLINK-20628] Port RabbitMQ Connector using unified Source

### PR DESCRIPTION
**Pull request by RocMarshal**
related pr #1

### CODE REVIEW

File: flink-connector-rabbitmq/src/main/java/org/apache/flink/connector/rabbitmq/common/ConsistencyMode.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * <ul>
+ *   <li><code>AT_MOST_ONCE</code> Messages are consumed by the output once or never.
+ *   <li><code>AT_LEAST_ONCE</code> Messages are consumed by the output at least once.
+ *   <li><code>EXACTLY_ONCE</code> Messages are consumed by the output exactly once.
+ * </ul>
+ *
+ * <p>Note that the higher the consistency guarantee gets, fewer messages can be processed by the
+ * system. At-least-once and exactly-once should only be used if necessary.
+ */
+public enum ConsistencyMode {
```
Comment (RocMarshal) [inline flink-connector-rabbitmq/src/main/java/org/apache/flink/connector/rabbitmq/common/ConsistencyMode.java:35]: Should we use `DeliveryGuarantee` to replace the `ConsistencyMode` ?

**Comment by MartijnVisser** [comment]
@RocMarshal Without actually seeing a passing CI, it doesn't make much sense to review this one...

**Comment by RocMarshal** [comment]
The new sink https://github.com/apache/flink-connector-rabbitmq/pull/29

**Comment by RocMarshal** [comment]
> @RocMarshal Without actually seeing a passing CI, it doesn't make much sense to review this one...

Got it.
I have re associated the corresponding PR information. And this pr will only do something about new source.

**Comment by RocMarshal** [comment]
@vahmed-hamdy will take over it(With discussion offline).
Thanks @vahmed-hamdy & @MartijnVisser ~ 
Looking forward to it !

**Pull request closed without merging**